dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/transport/DecodeHandler.java

```diff
@@ -38,19 +39,33 @@ public DecodeHandler(ChannelHandler handler) {
 
     @Override
     public void received(Channel channel, Object message) throws RemotingException {
+        doRecveived(message);
+
+
+        try {
+            handler.received(channel, message);
+        } catch (RetryHandleException e) {
+            if (message instanceof Request) {
+                Request request = (Request) message;
+                retry(request.getData());
+            } else {
+                // Retry only once, and only Request will throw an RetryHandleException
+                throw new RemotingException(channel, "Unknown error encountered when retry handle: " + e.getMessage());
+            }
+            handler.received(channel, message);
```

Comment:
Should 'retry once handler.received' be part of the retry implementation, rather than a separate public part.